### PR TITLE
getting jenkins root url from Jenkins class instance

### PR DESCRIPTION
**Pull request by tomascea**

### CODE REVIEW

**Comment by proski** [comment]
The broken link in Stash posts is the biggest annoyance with the new Jenkins. Thank you for fixing the issue. Please merge and release soon!

**Comment by jimklimov** [comment]
@tomascea : just in case, is your PR a sort of duplicate (alternative) to my #26 which I think it is? Just in case, ask someone from IRC to review it, because I also started the fixed logic as big and branchey, but in the end at https://github.com/jenkinsci/jenkins/pull/3570#issuecomment-423195479 was recommended to use the single global object getter, so the change to fix became a one-liner.

**Review by goostleek — COMMENTED** [review]
I think the #26 from @jimklimov fixes the same issue in a much more concise way.

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java [hunk 1]
```diff
@@ -49,15 +50,22 @@ public void onCompleted(AbstractBuild build, TaskListener listener) {
...
+        // JenkinsLocationConfiguration globalConfig = new JenkinsLocationConfiguration();
+        String rootUrl = "";
+        String buildUrl;
+        Jenkins jenkinsInstance = Jenkins.getInstance();
+        if (jenkinsInstance != null ){
+            rootUrl = jenkinsInstance.getRootUrl();
+            if (!rootUrl.isEmpty()){
+                buildUrl = rootUrl + build.getUrl();
+            }else{
+                buildUrl = "Jenkins instance root URL is empty "+ build.getUrl();
```
Comment (goostleek) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:62]: I think the previous message (`PLEASE SET JENKINS ROOT URL FROM GLOBAL CONFIGURATION`) is more valuable for the user. It says precisely to adjust Jenkins configuration to get the proper build URL.

**Pull request closed without merging**